awscli/argprocess.py

```diff
@@ -245,13 +246,41 @@ def _is_complex_shape(model):
     return True
 
 
+def _uses_old_list_case(service_name, operation_name, argument_name):
+    """
+    Determines whether a given operation for a service needs to use the
+    deprecated shorthand parsing case for lists of structures that only have
+    a single member.
+    """
+    if service_name == 'elb':
+        service_name = 'elasticloadbalancing'
+    cases = {
+        'firehose': {
+            'put-record-batch': ['records']
+        },
+        'workspaces': {
+            'reboot-workspaces': ['reboot-workspace-requests'],
+            'rebuild-workspaces': ['rebuild-workspace-requests'],
+            'terminate-workspaces': ['terminate-workspace-requests']
+        },
+        'elasticloadbalancing': {
```

Comment:
I would prefer if we that we key by service/command name rather than endpoint. There is no guarantee that endpoint prefixes are not the same for different clients/commands.